### ISSUE TITLE
Binaries with a dash char as part of their filenames fails to command-complete when using `clink.argmatcher()`

### ISSUE THREAD

**Issue by garoto** [chrisant996/clink#194]
Trying to port my previous simple man `youtube-dl` auto-complete _script_ and I can't get it to work. Snippet below.

```lua
clink.argmatcher("youtube-dl")
:addflags("--abort-on-error", "--abort-on-unavailable-fragment", "--add-header", "--add-metadata")
```

Any binary with a dash in its filename will fail the argmatcher if my simple testing isn't flawed.

> clink --version
1.2.43.0119c5

Running Windows 8.1.

Edit: adding my current clink_settings since they can be relevant perhaps:

```
clink.autostart                   
clink.colorize_input              True
clink.default_bindings            bash
clink.gui_popups                  False
clink.logo                        none
clink.paste_crlf                  space
clink.path                        
clink.promptfilter                False
cmd.auto_answer                   answer_yes
cmd.ctrld_exits                   True
cmd.get_errorlevel                True
color.arg                         bold
color.arginfo                     yellow
color.argmatcher                  
color.cmd                         bold
color.cmdredir                    bold
color.cmdsep                      bright yellow
color.description                 bright green
color.doskey                      bold cyan
color.filtered                    bold
color.flag                        default
color.hidden                      
color.horizscroll                 
color.input                       
color.interact                    bright yellow on blue
color.message                     default
color.popup                       
color.popup_desc                  
color.prompt                      
color.readonly                    
color.selected_completion         
color.selection                   underline red
color.unexpected                  default
debug.log_terminal                False
doskey.enhanced                   False
exec.aliases                      True
exec.commands                     True
exec.cwd                          True
exec.dirs                         True
exec.enable                       True
exec.files                        False
exec.path                         True
exec.space_prefix                 True
files.hidden                      True
files.system                      False
history.dont_add_to_history_cmds  exit,ls,la,del,cls,vd,cd
history.dupe_mode                 erase_prev
history.expand_mode               off
history.ignore_space              True
history.max_lines                 0
history.save                      True
history.shared                    True
history.sticky_search             False
lua.break_on_error                False
lua.break_on_traceback            False
lua.debug                         False
lua.path                          
lua.reload_scripts                False
lua.strict                        True
lua.traceback_on_error            False
match.expand_envvars              True
match.ignore_accent               True
match.ignore_case                 relaxed
match.sort_dirs                   with
match.translate_slashes           system
match.wild                        True
prompt.async                      True
prompt.transient                  off
readline.hide_stderr              False
terminal.adjust_cursor_style      True
terminal.differentiate_keys       False
terminal.emulation                auto
terminal.raw_esc                  False
terminal.use_altgr_substitute     False
```

**Comment by chrisant996** [comment]
What does "Any binary with a dash in its filename will fail the argmatcher" mean?

In other words, what are the steps to reproduce the issue?  What does one need to type?

**Comment by garoto** [comment]
Paste the below snippet to a new .lua script that clink will read:

```lua
local my_parser = clink.argmatcher("yt-dlp")
:addarg("--add", "--status", "--commit", "--checkout")
```

Replace _"yt-dlp"_ with any binary that has a dash as part of its filename (rename a random executable if necessary).

**Comment by garoto** [comment]
```
[ R:\ ] > type C:\Apps\Diversos\Clink\blap.lua
clink.argmatcher("curl")
:addflags("--abort-on-error", "--abort-on-unavailable-fragment", "--add-header", "--add-metadata")

[ R:\ ] > curl -
--abort-on-error                 --abort-on-unavailable-fragment  --add-header                     --add-metadata

[ R:\ ] > type C:\Apps\Diversos\Clink\blap.lua
clink.argmatcher("curl-dash")
:addflags("--abort-on-error", "--abort-on-unavailable-fragment", "--add-header", "--add-metadata")

[ R:\ ] > curl-dash.exe - <TAB> 
```

Pressing TAB does nothing when the executable has a dash character as part of its filename.

**Comment by chrisant996** [comment]
I don't think this has anything to do with dashes.

What is `%PATHEXT%` set to?

**Comment by garoto** [comment]
> What is `%PATHEXT%` set to?

Windows defaults as far as Win8.1 goes:
```
> echo %PATHEXT%
.COM;.EXE;.BAT;.CMD;.MSC;.VBS;.VBE
```

**Comment by chrisant996** [comment]
```
> type \wbin\clink\blap.lua
clink.argmatcher("curl")
:addflags("--abort-on-error", "--abort-on-unavailable-fragment", "--add-header", "--add-metadata")

clink.argmatcher("curl-dash")
:addflags("--abort-on-error", "--abort-on-unavailable-fragment", "--add-header", "--add-metadata")

> curl -
--abort-on-error                 --add-header
--abort-on-unavailable-fragment  --add-metadata
> curl.exe -
--abort-on-error                 --add-header
--abort-on-unavailable-fragment  --add-metadata
> curl-dash -
--abort-on-error                 --add-header
--abort-on-unavailable-fragment  --add-metadata
> curl-dash.exe -
--abort-on-error                 --add-header
--abort-on-unavailable-fragment  --add-metadata
```

Settings are:
```
clink.autostart                   
clink.colorize_input              True
clink.default_bindings            bash
clink.gui_popups                  False
clink.logo                        short
clink.paste_crlf                  ampersand
clink.path
clink.promptfilter                True
cmd.auto_answer                   answer_yes
cmd.ctrld_exits                   False
cmd.get_errorlevel                True
color.arg                         bold
color.arginfo                     yellow
color.argmatcher                  bold green
color.cmd                         bold
color.cmdredir                    sgr 38;5;172
color.cmdsep                      sgr 38;5;214
color.description                 bright cyan
color.doskey                      bold cyan
color.filtered                    bold
color.flag                        default
color.git.star                    bright green
color.hidden                      bright red
color.horizscroll
color.input                       bright yellow
color.interact                    bold
color.luacode
color.luaprefix                   bright cyan on blue
color.message                     default
color.popup
color.popup_desc
color.prompt
color.readonly                    green
color.selected_completion
color.selection                   black on yellow
color.unexpected                  default
debug.log_terminal                False
doskey.enhanced                   True
exec.aliases                      True
exec.commands                     True
exec.cwd                          True
exec.dirs                         True
exec.enable                       True
exec.files                        True
exec.path                         False
exec.space_prefix                 True
files.hidden                      True
files.system                      False
history.dont_add_to_history_cmds  exit history
history.dupe_mode                 erase_prev
history.expand_mode               not_quoted
history.ignore_space              True
history.max_lines                 0
history.save                      True
history.shared                    False
history.sticky_search             False
lua.break_on_error                True
lua.break_on_traceback            False
lua.debug                         True
lua.path
lua.reload_scripts                False
lua.show_match_type               True
lua.strict                        True
lua.traceback_on_error            False
match.expand_envvars              True
match.ignore_accent               True
match.ignore_case                 relaxed
match.sort_dirs                   with
match.translate_slashes           system
match.wild                        True
prompt.async                      True
prompt.transient                  same_dir
readline.hide_stderr              False
terminal.adjust_cursor_style      True
terminal.differentiate_keys       False
terminal.emulation                auto
terminal.raw_esc                  False
terminal.use_altgr_substitute     False
```

**Comment by chrisant996** [comment]
Do you have any match generators defined in Lua scripts?
Maybe one is accidentally halting generators from running when a `-` is present in the first word?

**Comment by garoto** [comment]
> Do you have any match generators defined in Lua scripts?

`blap.lua` is the only .lua script I'm running right now. But perhaps I have more than a layer of stuff going on right now (consolez & ansicon) . Will try and troubleshoot further tomorrow, it kinda late on my tz.

**Comment by chrisant996** [comment]
Some ideas for troubleshooting:

1. does `curl.exe -`<kbd>Tab</kbd> work?  (Trying with .exe suffix and without dash in name.)
2. does `curl-dash -`<kbd>Tab</kbd> work?  (Trying with dash in name and without .exe suffix.)
3. run `clink info` and observe the listed script directories -- are there any other .lua files in any of them?

**Comment by chrisant996** [comment]
Found what's happening.

You and I both have `set completion-map-case on` in our `.inputrc` files.
It affects the dash.

But the `.inputrc` file is not loaded until _after_ the first time scripts are loaded.

If you use <kbd>Ctrl+X</kbd><kbd>Ctrl+R</kbd> to reload scripts, then the argmatchers will work.
When I was trying to reproduce the problem, I reloaded in order to load the new script.
And since only the very first load suffers from the problem, I was not able to reproduce the problem.

I'll figure out what to do about this.
